pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -75,51 +76,38 @@ func (lp *LifecycleProcessor) startExecutionSpan(event interface{}, rawPayload [
 // It should be called at the end of the invocation.
 func (lp *LifecycleProcessor) endExecutionSpan(endDetails *InvocationEndDetails) *pb.Span {
 	executionContext := lp.GetExecutionInfo()
-	duration := endDetails.EndTime.UnixNano() - executionContext.startTime.UnixNano()
+	start := executionContext.startTime.UnixNano()
+
+	traceID := executionContext.TraceID
+	spanID := executionContext.SpanID
+	// If we fail to receive the trace and span IDs from the tracer during a timeout we create it ourselves
+	if endDetails.IsTimeout && traceID == 0 {
+		traceID = random.Random.Uint64()
+		lp.requestHandler.executionInfo.TraceID = traceID
+	}
+	if endDetails.IsTimeout && spanID == 0 {
+		spanID = random.Random.Uint64()
+	}
 
 	executionSpan := &pb.Span{
 		Service:  "aws.lambda", // will be replaced by the span processor
 		Name:     "aws.lambda",
 		Resource: os.Getenv(functionNameEnvVar),
 		Type:     "serverless",
-		TraceID:  executionContext.TraceID,
-		SpanID:   executionContext.SpanID,
+		TraceID:  traceID,
+		SpanID:   spanID,
 		ParentID: executionContext.parentID,
-		Start:    executionContext.startTime.UnixNano(),
-		Duration: duration,
+		Start:    start,
+		Duration: endDetails.EndTime.UnixNano() - start,
 		Meta:     lp.requestHandler.triggerTags,
 		Metrics:  lp.requestHandler.triggerMetrics,
 	}
-	executionSpan.Meta["request_id"] = endDetails.RequestID
-	executionSpan.Meta["cold_start"] = fmt.Sprintf("%t", endDetails.ColdStart)
-	if endDetails.ProactiveInit {
-		executionSpan.Meta["proactive_initialization"] = fmt.Sprintf("%t", endDetails.ProactiveInit)
-	}
-	langMatches := runtimeRegex.FindStringSubmatch(endDetails.Runtime)
-	if len(langMatches) >= 2 {
-		executionSpan.Meta["language"] = langMatches[1]
-	}
-	captureLambdaPayloadEnabled := config.Datadog.GetBool("capture_lambda_payload")
-	if captureLambdaPayloadEnabled {
-		capturePayloadMaxDepth := config.Datadog.GetInt("capture_lambda_payload_max_depth")
-		requestPayloadJSON := make(map[string]interface{})
-		if err := json.Unmarshal(executionContext.requestPayload, &requestPayloadJSON); err != nil {
-			log.Debugf("[lifecycle] Failed to parse request payload: %v", err)
-			executionSpan.Meta["function.request"] = string(executionContext.requestPayload)
-		} else {
-			capturePayloadAsTags(requestPayloadJSON, executionSpan, "function.request", 0, capturePayloadMaxDepth)
-		}
-		responsePayloadJSON := make(map[string]interface{})
-		if err := json.Unmarshal(endDetails.ResponseRawPayload, &responsePayloadJSON); err != nil {
-			log.Debugf("[lifecycle] Failed to parse response payload: %v", err)
-			executionSpan.Meta["function.response"] = string(endDetails.ResponseRawPayload)
-		} else {
-			capturePayloadAsTags(responsePayloadJSON, executionSpan, "function.response", 0, capturePayloadMaxDepth)
-		}
-	}
```

Comment:
My request is that unless you have changed the logic here at all, that you don't make large refactors like this. It makes reviewing the code a lot harder. I find myself comparing line by line to see where the diffs are. That's annoying and I don't want to have to do it.

Reply:
Sorry! I'll look this over and see if this is still necessary and revert if not so it's easy to read through